File: ruby/hamming/hamming.rb
class Hamming
  VERSION = 3
  def self.compute(strand1, strand2)
    raise ArgumentError,
      "Strands must be of equal length" unless strand1.length == strand2.length
    strand1.chars.zip(strand2.chars).select { |n1, n2| n1 != n2 }.count
  end
end

File: javascript/hamming/hamming.js
var Hamming = function() {}

Hamming.prototype.compute = function(string1, string2) {
    if (string1.length !== string2.length)
        throw "DNA strands must be of equal length.";
    return string1.split("").filter(function(char, idx) {
        return char !== string2[idx]
    }).length
}

module.exports = Hamming;